pandas/tests/groupby/test_frame_value_counts.py

```diff
@@ -0,0 +1,254 @@
+    elif groupby == "column" or as_index:
+        # (otherwise SeriesGroupby crashes)
+        # compare against SeriesGroupBy value_counts
+        education_df["both"] = education_df["gender"] + "-" + education_df["education"]
+        expected = gp["both"].value_counts(
+            normalize=normalize, sort=sort, ascending=ascending
+        )
+        if as_index:
+            tm.assert_numpy_array_equal(result.values, expected.values)
+        else:
+            tm.assert_numpy_array_equal(
+                result[RESULT_NAME].values, expected[RESULT_NAME].values
+            )
+
+
+@pytest.mark.parametrize("normalize", [True, False])
+@pytest.mark.parametrize(
+    "sort, ascending",
+    [
+        (False, None),
+        (True, True),
+        (True, False),
+    ],
+)
+def test_compound(education_df, normalize, sort, ascending):
+    # Multiple groupby keys and as_index=False
+    gp = education_df.groupby(["country", "gender"], as_index=False, sort=False)
+    result = gp["education"].value_counts(
+        normalize=normalize, sort=sort, ascending=ascending
+    )
+
+    if sort:
+        # compare against apply with DataFrame value_counts
+        expected_values = gp.apply(
+            _frame_value_counts, "education", normalize, sort, ascending
+        ).values
+    elif normalize:
+        expected_values = np.array([1.0, 1.0 / 3, 1.0, 2.0 / 3, 1.0])
+    else:
+        expected_values = np.array([1, 1, 1, 2, 1], dtype=np.int64)
+
+    tm.assert_numpy_array_equal(result[RESULT_NAME].values, expected_values)
+
+
+@pytest.fixture
+def animals_df():
+    return pd.DataFrame(
+        {"num_legs": [2, 4, 4, 6], "num_wings": [2, 0, 0, 0]},
+        index=["falcon", "dog", "cat", "ant"],
+    )
+
+
+@pytest.mark.parametrize(
+    "sort, ascending, normalize, expected_data, expected_index",
+    [
+        (False, None, False, [1, 2, 1], [(2, 4, 6), (2, 0, 0)]),
+        (True, True, False, [1, 1, 2], [(2, 6, 4), (2, 0, 0)]),
+        (True, False, False, [2, 1, 1], [(4, 2, 6), (0, 2, 0)]),
+        (True, False, True, [0.5, 0.25, 0.25], [(4, 2, 6), (0, 2, 0)]),
+    ],
+)
+def test_data_frame_value_counts(
+    animals_df, sort, ascending, normalize, expected_data, expected_index
+):
+    # 3-way compare with :meth:`~DataFrame.value_counts`
+    # Tests from frame/methods/test_value_counts.py
+    result_frame = animals_df.value_counts(
+        sort=sort, ascending=ascending, normalize=normalize
+    )
+    expected = pd.Series(
+        data=expected_data,
+        index=pd.MultiIndex.from_arrays(
+            expected_index, names=["num_legs", "num_wings"]
+        ),
+    )
+    tm.assert_series_equal(result_frame, expected)
+
+    animals_df["key"] = 1
+
+    result_frame_groupby = animals_df.groupby("key").value_counts(
+        sort=sort, ascending=ascending, normalize=normalize
+    )
+    result_frame_groupby.reset_index(drop=True, level="key", inplace=True)
```

Comment:
don't use inplace in tests unless its a test about inplace

Reply:
pls compare the actual result of the test and don't manipulate the result, you *can* manipulate the expected

Reply:
Done